iree/turbine/kernel/ops/wave_ops.py

```diff
@@ -1380,6 +1388,36 @@ def is_contiguous_vec(self) -> bool:
         )
 
 
+@define_op("apply_expr")
+@dataclass
+class ApplyExpr(CustomOp):
+    register_: fx.Proxy
+    expr: Callable
+
+    @property
+    def type(self) -> "Register":
+        return get_custom(self.register_).type
+
+    @property
+    def indexing_dims(self) -> list[IndexSymbol]:
+        return get_custom(self.register_).indexing_dims
+
+
+@define_op("set_symbol")
+@dataclass
+class SetSymbol(CustomOp):
+    symbol: IndexExpr
+    register_: fx.Proxy
```

Comment:
Just wondering why mark these as type fx.Proxy instead of fx.Node?

Reply:
Half of the ops are using `fx.Proxy` intead `fx.Node`, we should probably align all of them.